python/social_bot/gazebo_agent.py

```diff
@@ -341,3 +352,74 @@ def get_contacts(self, contacts_sensor, contact_collision):
                     1] == contact_collision:
                 return True
         return False
+
+
+class ActionWrapper():
+    """ The action wrapper transform a new actions to primitive actions.
+
+    The primitive actions (like the force/velocity/position of joints) may have redundant
+    dimensions or offsets. By the action wrapper, we can transform the action to more
+    efficency one.
+    """
+
+    def get_actions_dim(self):
+        """ Get the dimension of the new action space
+        """
+        raise NotImplementedError("get_actions_dim not implemented!")
+
+    def warp_actions(self, action):
+        """ Wrap transformed actions to primitive actions.
+
+        Args:
+            action (nparray): the new action from policy network
+        Returns:
+            np.array, the primitive actions send to simulator
+        """
+        raise NotImplementedError("warp_actions not implemented!")
+
+@gin.configurable
+class YoubotActionWrapper(ActionWrapper):
+    """ This action wrapper transform a new actions to primitive actions.
+
+    The new action space is the same as keyboard demostration interface, defined as follows:
+    [
+        "arm_joint_yaw", "arm_joint_pitch", "arm_joint_pitch_2", "palm_joint",
+        "gripper_finger_joint", "wheel_speed", "wheel_turning"
+    ]
+    The primitive actions (the joints) please refer to social_bot/models/agent_cfg.json.
+    """
+
+    def get_actions_dim(self):
+        """ Get the dimension of the new action
+        """
+        return 7
```

Comment:
Might be better to change the hard-coded number of actions by 1) maintaining a list of actions and 2) calculating the length of the list as actions_dim ?

Reply:
Done